lib/top_hangman/io.rb

```diff
@@ -1,55 +1,78 @@
 module TopHangman
   class IO
-    class << self
-      def execute(game, renderer)
-        game.start
+    RUNNING = :running
+    STOPPED = :stopped
 
-        while game.state == Game::RUNNING
-          renderer.show_header(errors_count: game.current_round.errors_count)
+    def initialize
+      @state = RUNNING
+    end
+
+    def execute(game, renderer, file_manager)
+      game.start
+
+      while game.running?
+        renderer.show_header(errors_count: game.current_round.errors_count)
 
-          execute_round_loop(renderer, game) while game.current_round.state == TopHangman::Round::RUNNING
+        execute_round_loop(renderer, game, file_manager) while game.current_round.running? && @state == RUNNING
 
-          show_ending_round_message(renderer, game)
-          ask_for_new_round(renderer, game)
-        end
+        return unless @state == RUNNING
+
+        show_ending_round_message(renderer, game)
+        ask_for_new_round(renderer, game)
       end
+    end
+
+    def execute_round_loop(renderer, game, file_manager)
+      renderer.ask_for_guess
+      guess_letter = gets.chomp
+
+      return save_game(file_manager) if guess_letter == '--save'
+      return load_game(renderer, file_manager) if guess_letter == '--load'
+
+      game.play_round(guess_letter)
```

Comment:
mhmm this seems odd. `guess_letter` is not always the guess letter hahah. What about renaming the stuff like this?
```suggestion
      renderer.ask_for_next_prompt
      new_prompt = gets.chomp

      return save_game(file_manager) if new_prompt == '--save'
      return load_game(renderer, file_manager) if new_prompt == '--load'

      game.play_round(new_prompt)
```

Reply:
done!